src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs

```diff
@@ -241,16 +242,24 @@ internal static void ApplyDefaultAppConfiguration(HostBuilderContext hostingCont
             appConfigBuilder.AddJsonFile("appsettings.json", optional: true, reloadOnChange: reloadOnChange)
                     .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: reloadOnChange);
 
-            if (env.IsDevelopment() && env.ApplicationName is { Length: > 0 })
+            if (env.ApplicationName is { Length: > 0 })
             {
-                try
-                {
-                    var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
-                    appConfigBuilder.AddUserSecrets(appAssembly, optional: true, reloadOnChange: reloadOnChange);
-                }
-                catch (FileNotFoundException)
+                string applicationName = env.ApplicationName.Replace(Path.DirectorySeparatorChar, '_')
+                                                           .Replace(Path.AltDirectorySeparatorChar, '_');
+                appConfigBuilder.AddJsonFile($"{applicationName}.settings.json", optional: true, reloadOnChange: reloadOnChange)
+                        .AddJsonFile($"{applicationName}.settings.{env.EnvironmentName}.json", optional: true, reloadOnChange: reloadOnChange);
+
+                if (env.IsDevelopment())
                 {
-                    // The assembly cannot be found, so just skip it.
+                    try
+                    {
+                        var appAssembly = Assembly.Load(new AssemblyName(applicationName));
```

Comment:
Let's not change the behavior of this. Just use `env.ApplicationName` here (the unsanitized one):

```suggestion
                        var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
```